### ISSUE TITLE
Location QR Codes

### ISSUE THREAD

**Issue by tankerkiller125** [sysadminsmedia/homebox#9]
### What is the problem you are trying to solve with this feature?

Original issue hay-kot/homebox#879 by @NedPool

There are currently no way to label things like bins, cabinets, boxes, etc. as a location with a QR code in the real world.

### What is the solution you are proposing?

A QR code generation page for locations.

### Contributions

- [X] I have searched through existing issues and feature requests to see if my idea has already been proposed.
- [ ] If this feature is accepted, I would be willing to help implement and maintain this feature.
- [ ] If this feature is accepted, I'm willing to sponsor the development of this feature.

**Comment by icanc0** [comment]
I'm gonna try implementing this pr again: https://github.com/hay-kot/homebox/pull/543

**Comment by katosdev** [comment]
> I'm gonna try implementing this pr again: [hay-kot/homebox#543](https://github.com/hay-kot/homebox/pull/543)

Love it! Thanks for the support @icanc0

**Comment by katosdev** [comment]
Location QR codes are a thing already (CC @icanc0 ) 
To do this: 
1. On the location page, select the QR icon next to edit. You are generated a QR code. 
![image](https://github.com/sysadminsmedia/homebox/assets/7927609/1f1bc972-4e2b-4b8e-aa66-9001c111acd7)

![image](https://github.com/sysadminsmedia/homebox/assets/7927609/92aa54b6-9404-488b-8974-ae359c0aa860)

**Comment by icanc0** [comment]
Yea, the original intent was to recursively generate QR codes for a location, which needs a rehaul since the current QR code is entirely client-side.

**Comment by tankerkiller125** [comment]
The label generator portion does not yet exist, which would be the important bit I think.

Honestly, I can see some "processes" being built around the ability to scan location QR codes off labels that have been printed in the future.

**Comment by icanc0** [comment]
> The label generator portion does not yet exist, which would be the important bit I think.

Yeah, that's the main hurdle, you can do so many things with a label generator on the backend. Having a label customization system would be nice but it's a lot of work to design + implement.

**Comment by NedPool** [comment]
> Yeah, that's the main hurdle, you can do so many things with a label generator on the backend. Having a label customization system would be nice but it's a lot of work to design + implement.

I've got some experience with label/pdf generation in php... I haven't done anything in Go yet, but I can take a stab at it when I get some free time, if its not been picked up yet.

**Comment by mcarbonne** [comment]
Regarding this discussion [#35](https://github.com/sysadminsmedia/homebox/discussions/35) (and https://github.com/hay-kot/homebox/issues/696), I know merging items and locations is a long job, but it would also benefits for this issue. I think we should consider this and list all other issues that might be covered by this refactor.

**Comment by katosdev** [comment]
@icanc0 hey - Just wondered if you'd had any success on this one at all? 😄

**Comment by gab696** [comment]
I would find this useful too!
Also a feature available on other inventory tools like Inventree: QRcode relink or QRcode scan
This would be very useful for people like me who are switching from a custom inventory tool that already has a large number of items tagged
Thanks !

**Comment by tonyaellie** [comment]
Closing this as part of consolidation into https://github.com/sysadminsmedia/homebox/issues/436, for anyone who's put a bounty on this issue, please consider moving your bounty to that Epic.
